frontend/components/commons/sidebar-feedback-task/SidebarFeedbackTask.component.vue

```diff
@@ -1,21 +1,115 @@
 <template>
-  <div class="sidebar"></div>
+  <div class="sidebar__container">
+    <SidebarFeedbackTaskPanel
+      :visible-panel="currentMetric"
+      @close-panel="closePanel"
+    >
+      <component v-if="getProgressComponentName" :is="getProgressComponentName"
+    /></SidebarFeedbackTaskPanel>
+    <SidebarFeedbackTask
+      @on-click-sidebar-action="onClickSidebarAction"
+      :sidebar-items="sidebarItems"
+      :active-buttons="[currentMetric, currentMode]"
+      :expanded-component="currentMetric"
+    />
+  </div>
 </template>
 
 <script>
-export default {};
+export default {
+  data: () => ({
+    currentMetric: null,
+    currentMode: "annotate",
+  }),
+  computed: {
+    getProgressComponentName() {
+      return (
+        this.sidebarItems.find((item) => item.id === this.currentMetric)
+          ?.component || null
+      );
+    },
+  },
+  created() {
+    this.sidebarItems = [
+      {
+        id: "annotate",
+        tooltip: "Hand labeling",
+        icon: "hand-labeling",
+        group: "Mode",
+        action: "change-view-mode",
+        type: "non-expandable",
+        relatedMetrics: ["progress", "stats"],
+      },
+      {
+        id: "explore",
+        tooltip: "Exploration",
+        icon: "exploration",
+        group: "Mode",
+        action: "change-view-mode",
+        type: "non-expandable",
+        relatedMetrics: ["progress", "stats"],
+      },
+      {
+        id: "progress",
+        tooltip: "Progress",
+        icon: "progress",
+        action: "show-metrics",
+        group: "Metrics",
+        type: "expandable",
+        component: "FeedbackTaskProgress",
+      },
+      {
+        id: "stats",
+        tooltip: "Stats",
+        icon: "stats",
+        action: "show-metrics",
+        type: "expandable",
+        group: "Metrics",
+      },
+      {
+        id: "refresh",
+        tooltip: "Refresh",
+        icon: "refresh",
+        group: "Refresh",
+        type: "non-expandable",
+        action: "refresh",
+      },
+    ];
+  },
```

Comment:
1/ performance: change sidebarItems from list of object to object where keys is the mode
2/ don't pass business parameter to the stupid component : remove type and action from the differents items